test/jdk/javax/swing/JTableHeader/8016524/Bug8016524.java

```diff
@@ -0,0 +1,116 @@
+import javax.swing.table.JTableHeader;
+
+public class Bug8016524 {
+
+    private static JFrame frame;
+    private static JTable table;
+    private static JScrollPane scrollableTable;
+    private static JTableHeader header;
+    private static Point point;
+
+    // added so as to get the correct pixel value of the bottom border
+    public static final int X_OFFSET = 10;
+    public static final int Y_OFFSET = 1;
+
+    public static final int FRAME_HT = 300;
+    public static final int FRAME_WT = 300;
+    public static final int TABLE_COLS = 5;
+    public static final int TABLE_ROWS = 2;
+
+    private static final int WHITE_RGB = Color.WHITE.getRGB();
+
+    public static void main(String[] args) throws Exception {
+
+        //to keep track of header dimensions
+        final int header_dim[] = new int[2];
+        Robot robot = new Robot();
+        robot.setAutoDelay(20);
+
+        try {
+            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+             throw new RuntimeException("Unsupported LookAndFeel Class: ");
+        }
+
+        SwingUtilities.invokeAndWait(() -> {
+            table = new JTable(TABLE_ROWS, TABLE_COLS);
+            scrollableTable = new JScrollPane(table);
+            frame = new JFrame();
+            frame.getContentPane().add(scrollableTable);
+            frame.setSize(FRAME_WT, FRAME_HT);
+            frame.setLocationRelativeTo(null);
+            frame.pack();
+            frame.setVisible(true);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+            // retrieve JTableHeader coordinate position on screen (x,y)
+            point = table.getTableHeader().getLocationOnScreen();
+            // retrieve height and width of the table
+            header = table.getTableHeader();
+            header_dim[0] = header.getHeight();
+            header_dim[1] = header.getWidth();
+        });
+
+        robot.waitForIdle();
+
+        // to check mouse pointer position on screen
+        robot.mouseMove(point.x + X_OFFSET, point.y + header_dim[0] - Y_OFFSET);
+        robot.waitForIdle();
+
+        // get pixel color at lower left and lower right on the JTableHeader border
+        Color lowerLeft = robot.getPixelColor(point.x + X_OFFSET, point.y + header_dim[0] - Y_OFFSET);
+        Color lowerRight = robot.getPixelColor(point.x + header_dim[1] - X_OFFSET, point.y + header_dim[0] - Y_OFFSET);
+
+        // if pixel color is white then border not visible, throw Exception
+        if(lowerLeft.getRGB() == WHITE_RGB || lowerRight.getRGB() == WHITE_RGB)
+        {
+            throw new RuntimeException("JTableHeader Bottom Border not visible");
+        }
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e)
+        {
+            throw new RuntimeException(e);
+        }
+        frame.dispose();
```

Comment:
frame should be disposed irrespective of whether it fails or pass.
Also, it should be under EDT too.

Reply:
The test is not marked for mac only and I see it fails in my ubuntu locally.
Did you try running the test in CI system?

Reply:
@prsadhuk I have changed the test case to explicitly set the table and header background to white. Mach5 tests are passing on all platforms including linux now. Can you please let me know in case you are seeing any issues on Ubuntu locally, with the new test case? In order to mark the test as mac only should we specify`@requires (os.family=="mac") ` in jtreg tag section?

Reply:
I now see it fail locally in windows system also along with ubuntu. windows screenshot here where the border is white which is why it fails
![JTheaderBorder](https://user-images.githubusercontent.com/43534309/151753660-92d6021a-30d6-4215-97b0-7ab17ddb88d4.png)

Reply:
@prsadhuk As you suggested earlier, should we mark it as mac-only test case to prevent test case issues arising  from platform specific System LAF ?

Reply:
@prsadhuk I have tagged this particular test case as mac-only to prevent test case issues arising from platform specific System LAF.

Reply:
It seems there's a anomaly for JTableHeader border in native and jdk in Windows L&F..
The system table shows here doesn't show any line
![JTheaderBorder-win](https://user-images.githubusercontent.com/43534309/151911992-846f2f21-c939-482f-9ef5-746d07c0b410.png)

whereas in our SwingSet2 we have a light grey line
![image](https://user-images.githubusercontent.com/43534309/151912084-c9cd39a5-74de-4ef9-a925-6081e8c4d294.png)

Technically, it might be construed as a bug but I guess I will prefer a demarcation line between header and body.
It's upto you if you want to take a look at this in a separate issue or in this.

Reply:
@prsadhuk Thank you for looking into it. I think this fix will require a change in Window related L&F classes. I think it would help me if I track this as separate issue.